### PR TITLE
feat(fonts): create font context and apply it

### PR DESCRIPTION
**Pull request by Boyadjie**
## 🤔 Why do you want to make those changes?

We want to have a common default font when using the remotion preview.

## 🧑‍🔬 How did you make them?

I've created a `fontContext` and it's `fontProvider`.I used the provider on `RootLayout.tsx` and on `Video.tsx`. I've updated the code for the font-picker with the context and made the font selection persistent with a session storage. The management of the selection is made in a hook called `useSelectedFont`, that handle the default font, if there is query params and le font-picker choice.

## 🧪 How to check them?

- On the preview you can try choosing a font on the event page, it should be kept on the videos when you change pages.

### CODE REVIEW

File: src/app/utils/loadFont.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import {top250} from '../../data/fonts';
+import {useSearchParams} from 'next/navigation';
+
+// Without this type, loadFont retrun a Type error.
```
Comment (Slocaly) [inline src/app/utils/loadFont.ts:4]: retrun => return

File: src/context/fonts/FontProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import {FontContext} from './FontCountext';
+import {loadFont} from '@remotion/google-fonts/NotoSans';
+import {loadGoogleFont} from '../../app/utils/loadFont';
+
+interface FontProviderProps {
+	children: ReactNode;
+}
+
+export const FontProvider: React.FC<FontProviderProps> = ({children}) => {
+	const [selectedFont, setSelectedFont] = useState<string | undefined>(
```
Comment (Slocaly) [inline src/context/fonts/FontProvider.tsx:12]: On a un cas ou notre font peut être `undefined` ? pourquoi ne pas forcer la font par défaut (Noto Sans) dans ce cas la ?

File: src/app/utils/loadFont.ts (hunk 1 above)
Comment (Slashgear) [reply src/app/utils/loadFont.ts:4]:
```suggestion
// Without this type, loadFont return a Type error.
```

**Review by Slashgear — APPROVED** [review]

File: src/app/utils/loadFont.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import {top250} from '../../data/fonts';
+
+// Without this type, loadFont return a Type error.
```
Comment (CruuzAzul) [inline src/app/utils/loadFont.ts:3]: Je suis pas sûr que le commentaire soit très utile, on se doute que sans le type ça va retourner une erreur, je pense pas qu'il soit nécessaire 😊

**Review by CruuzAzul — APPROVED** [review]

**Pull request merged**